### PR TITLE
FIX: cancel editing should clear the composer

### PR DESCRIPTION
**Pull request by jjaffeux**

### CODE REVIEW

File: plugins/chat/assets/javascripts/discourse/components/chat-live-pane.hbs [hunk 1]
```diff
@@ -79,7 +79,6 @@
     {{#if (or @channel.isDraft @channel.isFollowing)}}
       <ChatComposer
         @sendMessage={{this.sendMessage}}
-        @onCancelEditing={{this.cancelEditing}}
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/chat-live-pane.hbs:82]: dead code

File: plugins/chat/assets/javascripts/discourse/components/chat-thread.hbs [hunk 2]
```diff
@@ -49,7 +49,6 @@
   {{else}}
     <ChatComposer
       @sendMessage={{this.sendMessage}}
-      @onCancelEditing={{this.cancelEditing}}
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/chat-thread.hbs:52]: dead code

File: plugins/chat/assets/javascripts/discourse/components/chat-composer.js [hunk 3]
```diff
@@ -191,14 +191,12 @@ export default Component.extend(TextareaTextManipulation, {
...
 
-    if (event.keyCode === 27) {
-      // keyCode for 'Escape'
+    if (event.key === "Escape") {
       if (this.composerService?.replyToMsg) {
         this.set("value", "");
         this.composerService?.setReplyTo(null);
         return false;
       } else if (this.composerService?.editingMessage) {
-        this.set("value", "");
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/chat-composer.js:201]: this is encapsulated in `this.cancelEditing()`

File: plugins/chat/assets/javascripts/discourse/components/chat-composer.js [hunk 4]
```diff
@@ -191,14 +191,12 @@ export default Component.extend(TextareaTextManipulation, {
       this.paneService?.editLastMessageRequested();
     }
 
-    if (event.keyCode === 27) {
```
Comment (jjaffeux) [inline plugins/chat/assets/javascripts/discourse/components/chat-composer.js:194]: saves the comment ;)

**Comment by jjaffeux** [comment]
I will be reworking composer very soon so this is not definitive solution, but at least we have tests for this now.

**Review by CvX — APPROVED** [review]

**Pull request merged**